base/threadingconstructs.jl

```diff
@@ -38,8 +38,7 @@ function _threadsfor(iter,lbody)
                 end
             end
             # run this thread's iterations
-            for i = f:l
-                local $(esc(lidx)) = Base.unsafe_getindex(r,i)
+            for $(esc(lidx)) = r[f:l]
```

Comment:
This works as well with Arrays, Linspaces and Tuples.

It could and should be implemented trivially for anything with a flat representation, which is a prerequisite for the threaded loop anyway.